### ISSUE TITLE
Inferring the remote domain

### ISSUE THREAD

**Issue by baiyanhuang** [fabiospampinato/vscode-open-in-github#25]
Given we could alway get the remote url from current git repo:
```
$ git remote -v
origin  git@github.com:fabiospampinato/vscode-open-in-github.git (fetch)
origin  git@github.com:fabiospampinato/vscode-open-in-github.git (push)
```

We should probably do auto detect of the domain, rather than manually specify
Currently I need to work with repos from both github, and gitlab, and also our company's own domain, I will have to manually switch `openInGitHub.github.domain` to make it work, which is pretty tedious.

Related code should be here: 
https://github.com/fabiospampinato/vscode-open-in-github/blob/8e0faff9871edcd9697d9d9047fce26717f7ba49/src/utils.ts/#L108-L128

**Comment by fabiospampinato** [comment]
I've just rewritten the extension. This seems a good idea.

PR welcome to update [this regex](https://github.com/fabiospampinato/vscode-open-in-github/blob/668649ad62d78da343a1bfffd935ff200ae6d4ff/src/git.ts/#L82) to extract the domain, and delete the [`openInGitHub.github.domain`](https://github.com/fabiospampinato/vscode-open-in-github/blob/668649ad62d78da343a1bfffd935ff200ae6d4ff/src/git.ts#L68) configuration key.

The regex should work with urls in these styles, and maybe more if more styles are possible:

```
https://github.com/fabiospampinato/vscode-open-in-github.git
git@github.com:fabiospampinato/vscode-open-in-github.git
```

**Comment by fabiospampinato** [comment]
Implemented in v2.2.0, thanks for the suggestion 👍

**Comment by baiyanhuang** [comment]
Thanks, Fabio, this works perfectly!!!
